marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/DocumentRowConverter.java

```diff
@@ -82,17 +120,97 @@ private Iterator<DocBuilder.DocumentInputs> readContentFromRow(String uri, Inter
 
         DocumentMetadataHandle metadata = DocumentRowSchema.makeDocumentMetadata(row);
         DocBuilder.DocumentInputs mainInputs = new DocBuilder.DocumentInputs(uri, bytesHandle, uriTemplateValues, metadata);
-        final boolean extractedTextExists = row.numFields() == 9;
-        if (extractedTextExists) {
-            String extractedText = row.getString(8);
+        final int extractedTextRow = getExtractedTextRow(row);
+        if (extractedTextRow > 0) {
+            String extractedText = row.getString(extractedTextRow);
             DocBuilder.DocumentInputs extractedTextDocument = Format.XML.equals(this.extractedTextFormat) ?
                 buildExtractedXmlDocument(uri, extractedText, metadata) :
                 buildExtractedJsonDocument(uri, extractedText, metadata);
             return Stream.of(mainInputs, extractedTextDocument).iterator();
         }
+
+        final int classifiedTextRow = getClassifiedTextRow(row);
+        if (classifiedTextRow > 0) {
+            String classificationText = row.getString(classifiedTextRow);
+            String format = row.getString(2);
+            if ("XML".equals(format)) {
+                mainInputs = appendDocumentClassificationToXmlContent(new String(row.getBinary(1), StandardCharsets.UTF_8), uri, classificationText, uriTemplateValues, metadata);
```

Comment:
Just paste `row.getBinary(1)` to the method, that avoids converting it into a String first.